R-packages/evalcast/R/get_covidhub_predictions.R

```diff
@@ -243,31 +245,38 @@ get_forecaster_predictions <- function(covidhub_forecaster_name,
                                  type = "character"),
                   data.table = FALSE,
                   showProgress = FALSE)
-    # Specifying the date conversion after significantly speeds up loading 
+    # Specifying the date conversion after significantly speeds up loading
     # (~3x faster) for some reason
     pred$target_end_date = as.Date(pred$target_end_date)
     pred$forecast_date = as.Date(pred$forecast_date)
-    
+
     pcards[[forecast_date]] <- pred %>%
       separate(.data$target,
                into = c("ahead", "incidence_period", NA, "inc", "response"),
                remove = TRUE) %>%
-      mutate(forecaster = covidhub_forecaster_name, 
+      mutate(forecaster = covidhub_forecaster_name,
              incidence_period = if_else(
                .data$incidence_period == "wk", "epiweek","day"),
              inc = if_else(.data$inc == "inc", "incidence", "cumulative"),
-             response = case_when(.data$response == "death" ~ "deaths", 
+             response = case_when(.data$response == "death" ~ "deaths",
                                   .data$response == "case" ~ "confirmed",
-                                  TRUE ~ "drop",),
-             signal = paste(.data$response, .data$inc, "num", sep="_"),
-             data_source = if_else(.data$response=="drop", "drop", "jhu-csse"),
+                                  .data$response == "hosp" ~ "hosp",
+                                  TRUE ~ "drop"),
+             data_source = case_when(.data$response == "deaths" ~ "jhu-csse",
+                                  .data$response == "confirmed" ~ "jhu-csse",
+                                  .data$response == "hosp" ~ "hhs",
+                                  TRUE ~ "drop"),
+             signal = case_when(
+               .data$data_source == "jhu-csse" ~ paste(.data$response, .data$inc, "num", sep="_"),
+               .data$data_source == "hhs" & .data$inc == "incidence" ~ "confirmed_admissions_covid_1d",
```

Comment:
Can you extract this chunk (and the similar one marked below) into a separate (testable) function?